subsys/bootloader/bl_storage/Kconfig

```diff
@@ -6,4 +6,13 @@
 
 config SECURE_BOOT_STORAGE
 	bool "Functions for accessing the bootloader storage"
-	depends on SECURE_BOOT_CRYPTO
+
+config SB_IMPLEMENTATION_ID
+	bool "Uniquely identifies the underlying immutable PSA RoT."
+
+config SB_IMPLEMENTATION_ID_VALUE
+	string "32 byte Implementation ID value to program into the provision.hex file."
+	default "Company X Device Z NSIB x.y.z __"
```

Comment:
TF-M seem to think should be sha256. The spec says bytestring, restricting this to ascii  is maybe not ok.

Reply:
I wonder what the cleanest way of having a product-specific 32 byte value would be ...

Reply:
Damn provisioning.py, could have had a header file where the user can override which header gets included for his app.

But I can't think of an easy way for provisioning.py to read that header file.

Reply:
Mayb we require the user to fork us for now.

Reply:
@vili-nordic : Any ideas?

Reply:
Maybe we could put the implemenation ID in NSIB .rodata. Then we don't get trouble with provisioning.py.

AFAIK this does not violate any PSA requirement. I put it in OTP because It intuitively seemed to belong in OTP.

Reply:
@vili-nordic : This approach okay with you?

Reply:
That sounds good to me.

Reply:
I ended up with provisioning it from .rotdata to UICR in the provisioning image.

Not pushed yet.

Reply:
pushed